### ISSUE TITLE
Unit test fails due to apparent floating point arithmetic in the `moto` library

### ISSUE THREAD

**Issue by jtherrmann** [ASFHyP3/hyp3#2084]
While working on configurable credit costs, I added a unit test whose [final assertion](https://github.com/ASFHyP3/hyp3/pull/2072/files#diff-fbbb4d2d5125766993fc7e2406dfcb569624cd6989266d5ee5cb4da1d01e8165R383) fails with:

```
{'user_id': 'user1', 'remaining_credits': Decimal('11.700000000000003')} != {'user_id': 'user1', 'remaining_credits': Decimal('11.7')}
```

I confirmed that the `cost` passed to `decrement_credits` is `Decimal('88.3')`. I was unable to replicate this behavior using `boto3`'s `update_item` operation on a real DynamoDB table. It seems that there may be a bug in the `moto` library in which floating point arithmetic is used instead of decimal calculations.

**Comment by jtherrmann** [comment]
Unless someone sees something obvious that I'm missing, I think the next steps are:

* Find some workaround to make that assertion pass for now (may have to use a fuzzy comparison for `remaining_credits`)
* Create a minimal reproducible example using `moto` and `pytest` and open an issue at https://github.com/getmoto/moto/issues

**Comment by jtherrmann** [comment]
I created a minimal example at https://github.com/jtherrmann/moto-issue and opened an issue at https://github.com/getmoto/moto/issues/7349

**Comment by jtherrmann** [comment]
I'd be curious to see if this issue occurs with Moto's standalone server: https://docs.getmoto.org/en/latest/docs/server_mode.html

**Comment by jtherrmann** [comment]
I've opened a PR into `moto` to fix the issue (WIP, but allows the failing test to pass).

**Comment by jtherrmann** [comment]
Fix has been merged and made available as part of `moto==5.0.3.dev5`.
